exchanges/persisted-fetch/src/sha256.ts

```diff
@@ -30,61 +30,56 @@ const sha256Browser = (bytes: Uint8Array): Promise<Uint8Array> => {
   });
 };
 
-let nodeCrypto;
-if (typeof window === 'undefined') {
+let nodeCrypto: Promise<typeof import('crypto')> | void;
+if (typeof window === 'undefined' && !webCrypto) {
+  // Indirect eval'd require/import to guarantee no side-effects in module scope
+  // (optimization for minifiers)
   try {
-    // Indirect eval/require to guarantee no side-effects in module scope
-    // (optimization for minifiers)
-    nodeCrypto = new Function('require', 'return require("crypto")')(require);
-  } catch (e) {}
+    nodeCrypto = Promise.resolve(new Function('return require("crypto")')());
```

Comment:
Have you actually seen this work from a file?

If I go in the node repl, `nodeCrypto = new Function('return require("crypto")'` works. But from a file just like above I never get this to actually successfully do the `require`.

Reply:
Oh, yea, looks like in some environments `require` isn't available in the global scope. I was doubting this for a bit, hence the change 🤔